crates/bevy_dev_tools/src/lib.rs

```diff
@@ -0,0 +1,12 @@
+#![warn(missing_docs)]
```

Comment:
This can be moved to the [lints] table in Cargo.toml now :)

Reply:
I believe this would need to be in the main Cargo.toml, and some crates are not fully documented. This is because we use lints from the workspace. We could also just repeat what has been set in the main Cargo.toml. 

I'd leave it as is because it's simpler and consistent with how it's done in other crates.

Reply:
This can be configured per each crate. ([See announcement](https://blog.rust-lang.org/2023/11/16/Rust-1.74.0.html#lint-configuration-through-cargo)) If other crates don't use the `[lints]` table, then perhaps they need to be changed as well.

Reply:
Yep, we should migrate the other crates over too. Different PR though!

Reply:
Yes, but we are using workspace lints https://github.com/bevyengine/bevy/blob/8a523de8db28bfdc5b2a4c67c2022cc957075de6/Cargo.toml#L33-L41

And because of this, when we want to add lints only per crate we can't import lints from workspace because it results in: ```cannot override `workspace.lints` in `lints`, either remove the overrides or `lints.workspace = true` and manually specify the lints```

It's imported in  bevy_dev_tools/Cargo.toml via: 
```toml
[lints]
workspace = true
```

That's what I meant with:
> We could also just repeat what has been set in the main Cargo.toml.